Purchases/RCBackend.m

```diff
@@ -459,4 +482,80 @@ - (void)postOfferForSigning:(NSString *)offerIdentifier
                   }];
 }
 
+- (void)postSubscriberAttributes:(RCSubscriberAttributeDict)subscriberAttributes
+                       appUserID:(NSString *)appUserID
+                      completion:(nullable void (^)(NSError *_Nullable error))completion {
+    if (subscriberAttributes.count == 0) {
+        RCLog(@"called post subscriber attributes with an empty attributes dict!");
+        return;
+    }
+    NSString *escapedAppUserID = [self escapedAppUserID:appUserID];
+    NSString *path = [NSString stringWithFormat:@"/subscribers/%@/attributes", escapedAppUserID];
+    NSDictionary *attributesInBackendFormat = [self subscriberAttributesByKey:subscriberAttributes];
+    [self.httpClient performRequest:@"POST"
+                               path:path
+                               body:@{
+                                   @"attributes": attributesInBackendFormat
+                               }
+                            headers:self.headers
+                  completionHandler:^(NSInteger status, NSDictionary *_Nullable response, NSError *_Nullable error) {
+                      [self handleSubscriberAttributesResultWithStatusCode:status
+                                                                  response:response
+                                                                     error:error
+                                                                completion:completion];
+                  }];
+
+}
+
+- (NSDictionary<NSString *, NSDictionary *> *)subscriberAttributesByKey:(RCSubscriberAttributeDict)subscriberAttributes {
+    NSMutableDictionary <NSString *, NSDictionary *> *attributesByKey = [[NSMutableDictionary alloc] init];
+    for (NSString *key in subscriberAttributes) {
+        attributesByKey[key] = subscriberAttributes[key].asBackendDictionary;
+    }
+    return attributesByKey;
+}
+
+- (void)handleSubscriberAttributesResultWithStatusCode:(NSInteger)statusCode
+                                              response:(nullable NSDictionary *)response
+                                                 error:(nullable NSError *)error
+                                            completion:(void (^)(NSError *_Nullable error))completion {
+
+    if (completion == nil) {
+        return;
+    }
+
+    if (error != nil) {
+        completion([RCPurchasesErrorUtils networkErrorWithUnderlyingError:error]);
+        return;
+    }
+    NSError *responseError = nil;
+
+    if (statusCode > 300) {
+        NSDictionary *extraUserInfo = [self attributesUserInfoFromResponse:response
```

Comment:
We have a bug here, both in iOS and Android. When using a wrong API key, the backend returns a 401. Since 401 is > 300 and < 500, we will consider that the backend received the attributes and we will mark them as synced